packages/data/src/redux-store/index.js

```diff
@@ -527,99 +519,82 @@ function mapResolvers( resolvers, selectors, store, resolversCache ) {
 			fulfill: resolver, // Add the fulfill method.
 		};
 	} );
+}
 
-	const mapSelector = ( selector, selectorName ) => {
-		const resolver = resolvers[ selectorName ];
-		if ( ! resolver ) {
-			selector.hasResolver = false;
-			return selector;
+/**
+ * Returns resolvers with matched selectors for a given namespace.
+ * Resolvers are side effects invoked once per argument set of a given selector call,
+ * used in ensuring that the data needs for the selector are satisfied.
+ *
+ * @param {Object} selectors      The current selectors to be modified.
+ * @param {Object} resolvers      Resolvers to register.
+ * @param {Object} store          The redux store to which the resolvers should be mapped.
+ * @param {Object} resolversCache Resolvers Cache.
+ */
+function mapSelectorsWithResolvers(
+	selectors,
+	resolvers,
+	store,
+	resolversCache
+) {
+	function fulfillSelector( resolver, selectorName, args ) {
```

Comment:
Do you think the fact that it's not "async" now could change the "timing" for this to be triggered. This is not something I'm certain of, so take this with the grain of salt (and I'm sure you'll correct me if I'm wrong) but could "async" function be triggered after the consequent calls (like appender to the JS event loop like a setTimeout by default)?

Reply:
This can't change the timing because async functions run synchronously until the first `await`. Just like
```js
new Promise( (resolve) => { console.log('here'); } );
```
which also runs the callback synchronously. And generators (very similar to async functions) also run sync until the first `yield`.

You can verify this in console with:
```js
(()=>{console.log(1);(async()=>{console.log(2);await 0;console.log(3)})();console.log(4)})()
```
-- it will print 1, 2, 4, 3, in that order.